### PR TITLE
feat: implement "my position" rFOX dashboard queries

### LINKED ISSUE #7115: Implement rFOX dashboard "my position" query
# Overview

Create a react query to fetch/compute the values in the red box:

image

# References and additional details

Initial spike in #7092 

### Pending reward balance

This is the reward in RUNE for the current epoch.

`currentEpochDistributionAmountRuneBaseUnit` will be fetched separately using incremental solution(s) from https://github.com/shapeshift/web/issues/6951#issuecomment-2164122182

Calculating total reward - this is accurate enough for display purposes, within a couple hundred base units (see https://github.com/shapeshift/rFOX/issues/58#issuecomment-2159630701 )

```typescript
export const REWARD_RATE = 1n * 10n ** 27n;
export const WAD = 1n * 10n ** 18n;
const secondsInEpoch = epochEndBlock.timestamp - previousEpochEndBlock.timestamp
const totalEpochReward = (REWARD_RATE / WAD) * secondsInEpoch
```

Calculating account earnings - the reward this account has received so far for the epoch:

```typescript
const [previousEarningsForAccount, currentEarningsForAccount] =
      await Promise.all([
        publicClient.readContract({
          address: ARBITRUM_RFOX_PROXY_CONTRACT_ADDRESS,
          abi: stakingV1Abi,
          functionName: "earned",
          args: [account as Address],
          blockNumber: previousEpochEndBlockNumber, // The end of the previous epoch
        }),
        publicClient.readContract({
          address: ARBITRUM_RFOX_PROXY_CONTRACT_ADDRESS,
          abi: stakingV1Abi,
          functionName: "earned",
          args: [account as Address],
          blockNumber: epochEndBlockNumber,
        }),
      ]);

    const epochEarningsForAccount =
      currentEarningsForAccount - previousEarningsForAccount;
```

Calculating pending reward balance:

```
const pendingRewardBalanceForAccountBaseUnit = (epochEarningsForAccount / totalEpochReward) * currentEpochDistributionAmountRuneBaseUnit
```

### Time in Pool

This is the total time since a zero staking balance for the account. Likely requires pulling the contract logs for the account and iterating back in time to the most recent zero staking balance.

### My Reward Balance

TBD

### Lifetime Rewards

This requires use of the query from "Pending reward balance", aggregating the rune rewards for every epoch up to the epoch before the current one, starting from the life of the contract.

# Acceptance Criteria

The following values can be produced by the query:
* pending reward balance
* time in pool
* my reward balance
* lifetime rewards

# Need By Date

_No response_

# Screenshots/Mockups

_No response_

# Estimated effort

_No response_

### PR DESCRIPTION
**Pull request by woodenfurniture**
## Description

Implements queries required to populate "my position" data on the rFOX dashboard.

- [x] `Pending Reward Balance` implemented with `useCurrentEpochRewardsQuery`
- [x] `Lifetime Rewards` implemented with `useLifetimeRewardsQuery`
- [x] `Time in Pool` implemented with `useTimeInPoolQuery`
- [x] `My Reward Balance` not required as it's identical to `Pending Reward Balance`

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

closes #7115

Low risk - this feature is not production facing yet. The impact of these values being wrong would be low as its only used for display purposes

> What protocols, transaction types or contract interactions might be affected by this PR?

## Testing

Not currently testable as it's not connected to anything.

### Engineering

Note we're using a lossy approach to computing epoch reward in `calcEpochRewardForAccountRuneBaseUnit` - this is intentional as the precision loss is minimal and greatly simplifies things and massively reduces the fetching required to figure this out (we'd need to aggregate all contract logs for all users in the client, or create another indexer to do this)

### CODE REVIEW

**Review by gomesalexandre — COMMENTED** [review]
Killer PR ser. Core logic sounds sane to me, happy to do a second pass on this guy when opened and get it in so we can start wiring "My Position" up! 🐐 

Very nice one on the simple queryKey/queryFn wrappers as well!

- `useTimeInPoolQuery` implementation looks sane
- `useLifetimeRewardsQuery` implementation looks sane 
- `useEpochHistoryQuery` correctly returns mock data for now with a TODO
- `useEarnedQuery` implementation looks sane
- `useCurrentEpochRewardsQuery` implementation looks sane

File: src/contracts/constants.ts [hunk 1]
```diff
@@ -34,3 +34,6 @@ export const THOR_ROUTER_CONTRACT_ADDRESS_ETHEREUM = '0xd37bbe5744d730a1d98d8dc9
 
 // RFOX on Arbitrum ERC1967Proxy contract address
 export const RFOX_PROXY_CONTRACT_ADDRESS = '0xac2a4fd70bcd8bab0662960455c363735f0e2b56'
+export const RFOX_CONTRACT_CREATION_BLOCK_NUMBER = 222913582n
```
Comment (woodenfurniture) [inline src/contracts/constants.ts:37]: this wont work now that `RFOX_PROXY_CONTRACT_ADDRESS` is an env var

File: src/pages/RFOX/hooks/useLifetimeRewardsQuery.ts [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    () => [
+      'lifetimeRewards',
+      {
+        stakingAssetAccountAddress,
+      },
+    ],
+    [stakingAssetAccountAddress],
+  )
+
+  const queryFn = useMemo(
```
[the remark this answers is not in the document]
Comment (woodenfurniture) [reply src/pages/RFOX/hooks/useLifetimeRewardsQuery.ts:31]: sounds good, after discussing it makes sense to do this the tanstack way so will get that done in a follow-up

File: src/contracts/constants.ts (hunk 1 above)
Comment (woodenfurniture) [reply src/contracts/constants.ts:39]: will follow up with a solve

**Review by 0xApotheosis — APPROVED** [review]
Sanity checked code only, logic seems sane.

Stamping for velocity of the back of gomes's review 👌

**Pull request merged**